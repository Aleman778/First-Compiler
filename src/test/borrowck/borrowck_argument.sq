// ignore

fn mutate(v: &mut i32) {
    
}


fn foo(v: i32) {
    mutate(v);
}


fn main() {
    let x: &i32 = &10;
    mutate(x);
}
