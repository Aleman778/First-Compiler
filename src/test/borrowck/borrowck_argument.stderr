error: mismatched types
  --> $DIR/borrowck_argument.sq:7:12
   |
LL |     mutate(v);
   |            ^ expected `&mut i32`, found `i32`
   |

error: mismatched types
  --> $DIR/borrowck_argument.sq:13:12
   |
LL |     mutate(x);
   |            ^ expected `&mut i32`, found `&i32`
   |

